### PR TITLE
Improper search results - Fixes #2697

### LINKED ISSUE #2697: Plone 5.1 delivers improper search results page when enabled from search box.
Goto 

https://www.plone-demo.info/documents-all-content

and check "only in current section" and search for "document" and click the "Search" button.

This opens a URL 

https://www.plone-demo.info/@@search?path=%2Fplone%2Fdocuments-all-content&SearchableText=document

and shows 10 search results.

screenshot-www plone-demo info-2019-01-16-11-50-40

When you click on "Relevance" or a different tab then Plolne displays 21 search items in 3 batches.
Even showing a batch navigation with 3 batches for 21 items with a batch size of 10 is just broken.

screenshot-www plone-demo info-2019-01-16-11-51-20

### PR DESCRIPTION
**Pull request by sureshvv**
1. Add path to form as hidden so javascript uses it
2. Display section if search is inside current section

### CODE REVIEW

**Comment by sureshvv** [comment]
I am already a contributor. May be a different email though.

**Comment by sureshvv** [comment]
I am already a contributor. Have signed agreement.

**Comment by tmassman** [comment]
The commits look good but merging is blocked due to missing of the contributor agreement. If you used a different email address for this, then please adjust it.

**Comment by sureshvv** [comment]
@tmassman I have updated email address. Says I need 1 approving review now.

**Comment by sureshvv** [comment]
@polyester review please!

File: Products/CMFPlone/browser/templates/search.pt [hunk 1]
```diff
@@ -186,7 +188,9 @@
                 </span>
               </div>
 
-
+              <h4 tal:condition="section">
```
Comment (ale-rt) [inline Products/CMFPlone/browser/templates/search.pt]: Translations attributes are missing here

File: Products/CMFPlone/browser/templates/search.pt [hunk 2]
```diff
@@ -34,11 +34,13 @@
           tal:define="portal context/@@plone_portal_state/portal;
                       first_call not:request/advanced_search|nothing;
                       st python:request.get('SearchableText', '');
+                      section python:request.get('path', '').split('/')[2:];
                       DateTime python:modules['DateTime'].DateTime;
                       navigation_root_url view/navroot_url;">
 
         <input type="hidden" name="sort_on" value="" />
         <input type="hidden" name="sort_order" value="" />
+	<input type="hidden" name="path" value="" tal:attributes="value request/form/path | nothing" />
```
Comment (ale-rt) [inline Products/CMFPlone/browser/templates/search.pt]: Indentation is not consistent

File: Products/CMFPlone/browser/search.py [hunk 3]
```diff
@@ -221,6 +221,11 @@ def navroot_url(self):
             self._navroot_url = state.navigation_root_url()
         return self._navroot_url
 
+    def current_section_title(self):
+        if 'path' in self.request:
+            return self.context.unrestrictedTraverse(self.request['path']).Title()
```
Comment (ale-rt) [inline Products/CMFPlone/browser/search.py]: Uhm... I do not like this unrestrictedTraverse, also what happens if self.request['path'] is not a valid path?
Maybe 
```
section = self.context.restrictedTraverse(self.request['path'], None)
if section:
    return section.Title()
```
is better

File: Products/CMFPlone/browser/templates/search.pt (hunk 1 above)
Comment (sureshvv) [reply Products/CMFPlone/browser/templates/search.pt]: Fixed. Pls. check.

File: Products/CMFPlone/browser/templates/search.pt (hunk 2 above)
Comment (sureshvv) [reply Products/CMFPlone/browser/templates/search.pt]: Fixed

File: Products/CMFPlone/browser/search.py (hunk 3 above)
Comment (sureshvv) [reply Products/CMFPlone/browser/search.py]: Fixed

**Comment by jensens** [comment]
- change log news entry is missing.
- contributor agreement state fuzzy...?

**Comment by jensens** [comment]
There was no activity on this for a while. I close this request for now. If you plan to work further on this please reopen and go on.

**Pull request closed without merging**